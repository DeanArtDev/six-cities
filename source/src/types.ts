export enum AppState {
  PENDING = `pending`,
  READY = `ready`,
  LOADING = `loading`,
  SENT = `sent`,
}
export enum ErrorType {
  FAVORITE = `favorite`,
  UPLOAD_COMMENT = `uploadComment`,
  SING_IN = `singIn`,
}
export enum Authorization {
  AUTH = `AUTH`,
  NOT_AUTH = `NOT_AUTH`,
}
export enum PlacesClass {
  CITIES_LIST = `cities__places-list`,
  NEAR_LIST = `near-places__list`,
  CITIES_CARD = `cities__place-card`,
  CITIES_WRAPPER = `cities__image-wrapper`,
  NEAR_CARD = `near-places__card`,
  NEAR_WRAPPER = `near-places__image-wrapper`,
}
export enum OffersSortTypes {
  POPULAR = `Popular`,
  LOW_TO_HIGH = `Price: low to high`,
  HIGH_TO_LOW = `Price: high to low`,
  RATED_FIRST = `Top rated first`,
}
export enum NameSpace {
  MAIN = `MAIN`,
  USER = `USER`,
  DATA = `DATA`,
}
export enum FavoriteStatus {
  ON = `1`,
  OFF = `0`,
}
export enum ICON_SIZE {
  WIDTH = 30,
  HEIGHT = 40
}

export type Offer = {
  id: number,
  title: string,
  typeHousing: string,
  price: number,
  isPremium: boolean,
  features: Array<string>,
  maxAdults: number,
  bedrooms: number,
  isFavorite: boolean,
  host: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
  city: {
    coordinates: Array<number>,
    zoom: number,
    name: string,
  },
  location: {
    coordinates: Array<number>,
    zoom: number,
  },
  description: string,
  rating: number,
  images: Array<string>,
  prevImage: string,
  isProStatus: boolean,
}
export type Errors = {
  [key: string]: string | null,
}
export type AuthInfo = {
  id: number,
  email: string,
  name: string,
  avatarUrl: string,
  isProp: boolean,
}
export type SettingMap = {
  cityCoordinates: Array<number>,
  zoom: number,
}
export type CommentType = {
  id: number,
  rating: number,
  date: string,
  comment: string,
  user: {
    avatarUrl: string,
    name: string,
    id: number,
    isPro: boolean,
  },
}
export type FavoriteOffers = {
  [key: string]: Array<Offer>,
}
export type Id = number | null
