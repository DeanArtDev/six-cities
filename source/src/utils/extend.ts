const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
