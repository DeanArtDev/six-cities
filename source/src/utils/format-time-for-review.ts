import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
