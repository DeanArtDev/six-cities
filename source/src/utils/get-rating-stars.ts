const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
