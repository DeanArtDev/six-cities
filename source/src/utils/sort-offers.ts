import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
