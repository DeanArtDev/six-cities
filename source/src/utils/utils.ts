export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
