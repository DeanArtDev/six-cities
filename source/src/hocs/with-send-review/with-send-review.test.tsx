import withSendReview from './with-send-review';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {AppState, Authorization} from '@root/types';
import {noop} from '@root/utils/utils';


describe(`<withSendReview/>: `, () => {

  const WrappedComponent = React.forwardRef(
      function mockComponent(props, ratingRef: React.LegacyRef<HTMLElement>) {
        return (
          <section ref={ratingRef}/>
        );
      }
  );

  const MockWrapper = withSendReview(WrappedComponent);

  describe(`App State: `, () => {
    it(`should render READY correctly`, () => {
      const wrapper = renderer
       .create(
           <MockWrapper
             reviewValue={`Yes`}
             onValueChange={noop}
             onFormSubmit={noop}
             ratingRef={{}}
             detailId={333}
             appState={AppState.READY}
             authStatus={Authorization.AUTH}
           />,
           {createNodeMock: () => document.createElement(`section`)}
       ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
    it(`should render LOADING correctly`, () => {
      const wrapper = renderer
       .create(
           <MockWrapper
             reviewValue={`Yes`}
             onValueChange={noop}
             onFormSubmit={noop}
             ratingRef={{}}
             detailId={333}
             appState={AppState.LOADING}
             authStatus={Authorization.AUTH}
           />,
           {createNodeMock: () => document.createElement(`section`)}
       ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
    it(`should render SENT correctly`, () => {
      const wrapper = renderer
       .create(
           <MockWrapper
             reviewValue={`Yes`}
             onValueChange={noop}
             onFormSubmit={noop}
             ratingRef={{}}
             detailId={333}
             appState={AppState.SENT}
             authStatus={Authorization.AUTH}
           />,
           {createNodeMock: () => document.createElement(`section`)}
       ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
