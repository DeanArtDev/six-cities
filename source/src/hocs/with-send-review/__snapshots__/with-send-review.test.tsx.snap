// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withSendReview/>:  App State:  should render LOADING correctly 1`] = `<section />`;

exports[`<withSendReview/>:  App State:  should render READY correctly 1`] = `<section />`;

exports[`<withSendReview/>:  App State:  should render SENT correctly 1`] = `<section />`;
